### ISSUE TITLE
Not doing anything - web view shows 0 vilages

### ISSUE THREAD

**Issue by ilbinek** [stefan2200/TWB#51]
Hey there, so I got this up and running on a remote VPS and on localhost (WSL), to test it's not caused by my server, did not run simultaneously. 

The problem I got is that it even tells me it found my villages in the bot output, but then does nothing.
The web server also shows nothing when I click on village configuration.

Editing the config through web does edit the config in the bot.

Playing on CS server

Any help would be appreciated.

**Comment by suchencjusz** [comment]
Hi, try to edit the config file by notepad and run twb.py

**Comment by stefan2200** [comment]
I think this issue is caused by the web interface not able to locate the config file and cache directories `c_path = os.path.join("../cache", cache_location)`. Can you verify whether this issue is also present when running the bot in a screen session with something like:
```
screen -d -m python3 twb.py
screen -d -m bash -c "cd webmanager; python3 server.py"
```

**Comment by ilbinek** [comment]
> I think this issue is caused by the web interface not able to locate the config file and cache directories `c_path = os.path.join("../cache", cache_location)`. Can you verify whether this issue is also present when running the bot in a screen session with something like:
> 
> ```
> screen -d -m python3 twb.py
> screen -d -m bash -c "cd webmanager; python3 server.py"
> ```

Ran it using screen, same outcome. It's important to note that `cache/villages` is empty. Even when this is in console
```Village 22354 was found but no config entry was found. Adding automatically
Deployed new configuration file
Village 22357 was found but no config entry was found. Adding automatically
Deployed new configuration file```

Also `World map` in the browser does not work, wanted to create a separate issue about it, but my guess now is that it does not have the data needed from the bot.

I checked permissions, `cache` and all `cache/*` folders have `drwxrwxr-x`.

**Comment by ilbinek** [comment]
OK, another update. Running the `twb.py` in VSC (remote connection to WSL) in debug mode made the bot start doing things. cache folder now contains jsons.

Web view yields the same result when ran in this way.

The start commands in VSC are as follows
```
/usr/bin/env /bin/python3 /home/ilbinek/.vscode-server/extensions/ms-python.python-2023.14.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 51271 -- /home/ilbinek/TWB/twb.py

cd /home/ilbinek/TWB/webmanager ; /usr/bin/env /bin/python3 /home/ilbinek/.vscode-server/extensions/ms-python.python-2023.14.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 34433 -- /home/ilbinek/TWB/webmanager/server.py
```

**Comment by stefan2200** [comment]
Since you are using WSL, could you try if the issue exists in Windows as well?

**Comment by ilbinek** [comment]
I am not able to get Python and pip running on the Windows os itself.

But the problem also happens on my VPS.
Both VSP and WSL are running:
```
Ubuntu 22.04.3 LTS
Python 3.10.12
pip 22.0.2
```

I now tried to run it again on WSL using normal means after running it for some time in VSC. So far it looks like it's working, will update you.

**Comment by stefan2200** [comment]
Alright, I have no idea what I was thinking at the time, hard-coding slashes and stuff but I think I've found the issue. Could you check if the issue is fixed on this version? [Branch](https://github.com/stefan2200/TWB/tree/fix-webmanager-paths)

**Comment by ilbinek** [comment]
Alright, so I tested this branch. I'll reiterate the issue.

The bot does nothing, no files are created in the `cache` directory. The webmaster not working then is just because the bot does not provide the data into the cache directory.

This issue persists on the fix-webmanager-paths branch.

After getting the bot to run by "debugging" it in remote VSC connection, normal starting works.

It's probably something more with the first launch. Deleting the jsons in `cache` did not prevent me from starting it again and them being generated. So I would look elsewhere. I don't have tons of time rn, but I could give it more testing sometime next week.

**Comment by stefan2200** [comment]
I made some additional changes, fixing an old bug along the way. Could you check if [this branch](https://github.com/stefan2200/TWB/tree/new-dev)

**Comment by ilbinek** [comment]
Same result, 
![image](https://github.com/stefan2200/TWB/assets/23197934/47a8e1f9-a768-4acd-a24a-3c89d6ee84f5)
while cache folders are empty.
![image](https://github.com/stefan2200/TWB/assets/23197934/a67ff4dd-0383-433b-84d2-ff2f602a268e)
while reports should be around 1.3MB

**Comment by grebbekevin** [comment]
Hi guys,

I have got the same issue. The fix for me was to recheck the configurations of the config.json file.

I had the following code:
"world": {
    "knight_enabled": true,
    "flags_enabled": true,
    "quests_enabled": true,
    "trade_for_premium": false,
    "archers_enabled": true,
    "building_destruction_enabled": true,
    "boosters_enabled": true
  },


But the problem there was that I did not have knights in this world so the code would not run correctly.
If you verify your file line by line to check if you have those things on that world it should be working properly.

Kind regards,

Kevin
